### PR TITLE
[FIX] web_editor: properly handle nested lists and list conversion

### PR DESCRIPTION
**Pull request by Zynton**
These are various fixes for bugs concerning nested lists and the conversion a type of list into an other. Many tests were added to better cover nested lists as well as checklists in general.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Zinston** [comment]
@pparidans Thanks for the review. I applied your requested changes.

File: addons/web_editor/static/tests/wysiwyg_tests.js [hunk 1]
```diff
@@ -2529,11 +2873,70 @@ QUnit.test('checklist', function (assert) {
                         '<p>y</p>',
                 },
             },
+            {
```
Comment (pparidans) [inline addons/web_editor/static/tests/wysiwyg_tests.js]: Broken test.
This test should be moved to another commit (not fixed in the current one)

File: addons/web_editor/static/tests/wysiwyg_keyboard_tests.js [hunk 2]
```diff
@@ -2079,6 +2079,30 @@ var keyboardTestsComplex = [{
...
+            start: "p:eq(2)->0",
+            key: 'BACKSPACE',
+        }],
+        test: {
+            content: '<ul><li><p>1</p></li><li class="o_indent"><ul><li><p>2</p></li></ul></li><li><p>3</p></li></ul>',
+            start: "p:eq(1):contents()[0]->1",
+        },
+    },
+    {
+        name: "in ul.list-group (div > p > textbefore): BACKSPACE at start",
```
Comment (pparidans) [inline addons/web_editor/static/tests/wysiwyg_keyboard_tests.js]: Broken test in the current commit.

Comment (pparidans) [reply addons/web_editor/static/tests/wysiwyg_keyboard_tests.js]: Also `ul.list-group (div > p > textbefore): BACKSPACE at start (carret position)` test is also broken.

**Comment by Zinston** [comment]
@pparidans Thanks, all fixed!

**Comment by pparidans** [comment]
@robodoo r+

**Comment by pparidans** [comment]
@robodoo rebase-ff

**Comment by pparidans** [comment]
@robodoo retry

**Comment by pparidans** [comment]
@robodoo retry

**Comment by pparidans** [comment]
@robodoo retry

**Comment by pparidans** [comment]
@robodoo retry
